packages/ddp-client/test/livedata_tests.js

```diff
@@ -1201,6 +1201,32 @@ testAsyncMulti('livedata - methods with nested stubs', [
   },
 ]);
 
+Tinytest.addAsync('livedata - method interaction with unblocking mechanism', async function(test) {
+  let serverEvents = [];
+
+  // server-only methods
+  if (Meteor.isServer) {
+    Meteor.methods({
+      [`unblockedMethod${test.runId()}`]:  async function() {
+        serverEvents.push('unblock start');
+        this.unblock();
+        await Meteor._sleepForMs(2000);
+        serverEvents.push('unblock end')
+      },
+      [`blockingMethod${test.runId()}`]: async function() {
+        serverEvents.push('blockingMethod');
+      }
+    });
+  }
+
+  Meteor.callAsync(`unblockedMethod${test.runId()}`);
+  Meteor.callAsync(`blockingMethod${test.runId()}`);
+
+  if (Meteor.isServer) {
+    test.equal(serverEvents, ['unblock start', 'blockingMethod']);
+  }
+});
```

Comment:
@zodern: Does it seems correct this test for you according on what you commented [here](https://github.com/meteor/meteor/pull/13054#discussion_r1514693752)? 

I ask because even with the current version and doing the finish=true after serverPromise resolution, the test is passing green. Or maybe is a matter on the test itself and how it is run by server and client on distinct times.

Reply:
It would probably be easier to write the test in https://github.com/meteor/meteor/tree/059d55d7dd3f8335a87de49d8662df5c28d72814/packages/ddp-client/test/async_stubs

You can see how the other tests there handle queueing some methods, and then checking the server events.

Reply:
Done. I will ensure the change doesn't break other tests.

Please review.